### PR TITLE
fix: [#4204] Fix remaining eslint warnings - botbuilder-lg

### PR DESCRIPTION
**Pull request by erquirogasw**
Addresses # 4204
#minor

## Description
This PR fixes the ESLint and JSDoc issues in the botbuilder-lg library.

## Specific Changes
- Applied auto-fixes with yarn lint --fix.
	- Add trailing commas
	- Fix spacing.
	- Fix indentation.
	- Add blank line in between comments.
	- Fix quoting.
	- Replace arrow functions: () => with function ().
	- Replace let with const.
- Applied manual fixes for:
	- Missing documentation.
	- Unused parameters and imports.
	- Replacing hasOwnProperty calls.
	- Fixing lodash imports.

## Testing
image

image

### CODE REVIEW

File: libraries/botbuilder-lg/tsconfig.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "lib": ["es2015", "dom"],
     "outDir": "lib",
     "rootDir": "src",
-    "skipLibCheck": true
+    "skipLibCheck": true,
```
Comment (ceciliaavila) [inline libraries/botbuilder-lg/tsconfig.json:7]: Is this change necessary?

File: package.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
     "dev:link": "wsrun --if is-not-private --bin=yarn link",
     "dev:unlink": "wsrun --if is-not-private --bin=yarn unlink",
     "functional-test": "yarn build && yarn workspace functional-tests test",
-    "lint": "wsrun -m -l lint --quiet",
+    "lint": "wsrun -m -l lint",
```
Comment (ceciliaavila) [inline package.json:22]: Don't change this until the last PR.

File: yarn.lock [hunk 3]
```diff
@@ -3898,10 +3898,10 @@ chrome-trace-event@^1.0.2:
   dependencies:
     tslib "^1.9.0"
 
-chromedriver@^98.0.0:
-  version "98.0.1"
-  resolved "https://registry.yarnpkg.com/chromedriver/-/chromedriver-98.0.1.tgz#ccb1e36a003b4c6af0b184caa00fca8370d88f2a"
-  integrity sha512-/04KkHHE/K/lfwdPTQr5fxi1dWvM83p8T/IkYbyGK2PBlH7K49Dd71A9jrS+aWgXlZYkuHhbwiy2PA2QqZ5qQw==
+chromedriver@^100.0.0:
```
Comment (ceciliaavila) [inline yarn.lock:3901]: I think there's no need to commit this change.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]
Also see why the test:compat check is failing.

File: libraries/botbuilder-lg/src/evaluator.ts [hunk 4]
```diff
@@ -294,7 +300,9 @@ export class Evaluator extends AbstractParseTreeVisitor<unknown> implements LGTe
 
     /**
      * Visit a parse tree produced by the ifElseBody labeled alternative in LGTemplateParser.body.
+     *
      * @param ctx The parse tree.
+     * @returns the visitor result
```
Comment (sw-joelmut) [inline libraries/botbuilder-lg/src/evaluator.ts:305]:
```suggestion
     * @returns The visitor result.
```

File: libraries/botbuilder-lg/src/expander.ts [hunk 5]
```diff
@@ -93,7 +94,8 @@ export class Expander extends AbstractParseTreeVisitor<unknown[]> implements LGT
         this.templateMap = keyBy(templates.allTemplates, (t: Template): string => t.name);
         this.lgOptions = opt;
 
-        // generate a new customzied expression parser by injecting the template as functions
+        // generate a new customized expression parser by injecting the template as functions
+        // expressionParser: Given expression parser.
```
Comment (sw-joelmut) [inline libraries/botbuilder-lg/src/expander.ts:98]: Why this comment is necessary?

File: libraries/botbuilder-lg/src/expander.ts [hunk 6]
```diff
@@ -160,7 +165,9 @@ export class Expander extends AbstractParseTreeVisitor<unknown[]> implements LGT
 
     /**
      * Visit a parse tree produced by the ifElseBody labeled alternative in LGTemplateParser.body.
+     *
      * @param ctx The parse tree.
+     * @returns The result of visiting if-else body
```
Comment (sw-joelmut) [inline libraries/botbuilder-lg/src/expander.ts:170]:
```suggestion
     * @returns The result of visiting if-else body.
```

File: libraries/botbuilder-lg/src/expander.ts [hunk 7]
```diff
@@ -367,7 +376,9 @@ export class Expander extends AbstractParseTreeVisitor<unknown[]> implements LGT
 
     /**
      * Visit a parse tree produced by LGTemplateParser.normalTemplateString.
+     *
      * @param ctx The parse tree.
+     * @returns The result of visiting NormalTemplateString
```
Comment (sw-joelmut) [inline libraries/botbuilder-lg/src/expander.ts:381]:
```suggestion
     * @returns The result of visiting NormalTemplateString.
```

File: libraries/botbuilder-lg/src/templateExtensions.ts [hunk 8]
```diff
@@ -105,7 +108,9 @@ export class TemplateExtensions {
 
     /**
      * If a value is pure Expression.
+     *
      * @param ctx Key value structure value context.
+     * @returns true if the value is pure Expression, false otherwise.
```
Comment (sw-joelmut) [inline libraries/botbuilder-lg/src/templateExtensions.ts:113]: Start with uppercase and finish with a dot, multiple cases in others

**Review by sw-joelmut — CHANGES_REQUESTED** [review]
The changes in yarn.lock, package.json, tsconfig.json and .eslintrc.json should not be required

File: libraries/botbuilder-lg/tsconfig.json [hunk 9]
```diff
@@ -3,8 +3,7 @@
   "compilerOptions": {
     "lib": ["es2015", "dom"],
     "outDir": "lib",
-    "rootDir": "src",
-    "skipLibCheck": true
+    "rootDir": "src"
```
[the remark this answers is not in the document]
Comment (erquirogasw) [reply libraries/botbuilder-lg/tsconfig.json:6]: image

I'm seeing the setting in the branch. I'll review in git.

**Review by sw-joelmut — APPROVED** [review]

**Comment by erquirogasw** [comment]
@ceciliaavila Feedback applied. Thanks!

**Review by ceciliaavila — APPROVED** [review]

**Pull request closed without merging**